x-pack/legacy/plugins/siem/public/containers/anomalies/anomalies_query_tab_body/index.tsx

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { EuiSpacer } from '@elastic/eui';
+import { AnomaliesQueryTabBodyProps } from './types';
+import { manageQuery } from '../../../components/page/manage_query';
+import { AnomaliesOverTimeHistogram } from '../../../components/anomalies_over_time';
+import { AnomaliesOverTimeQuery } from '../anomalies_over_time';
+import { getAnomaliesFilterQuery } from './utils';
+import { useSiemJobs } from '../../../components/ml_popover/hooks/use_siem_jobs';
+import { useKibanaUiSetting } from '../../../lib/settings/use_kibana_ui_setting';
+import { DEFAULT_ANOMALY_SCORE } from '../../../../common/constants';
+
+const AnomaliesOverTimeManage = manageQuery(AnomaliesOverTimeHistogram);
+
+export const AnomaliesQueryTabBody = ({
+  endDate,
+  skip,
+  startDate,
+  type,
+  narrowDateRange,
+  filterQuery,
+  anomaliesFilterQuery,
+  setQuery,
+  hideHistogramIfEmpty,
+  updateDateRange = () => {},
+  AnomaliesTableComponent,
+  flowTarget,
+  ip,
+}: AnomaliesQueryTabBodyProps) => {
+  const [siemJobsLoading, siemJobs] = useSiemJobs(true);
+  const [anomalyScore] = useKibanaUiSetting(DEFAULT_ANOMALY_SCORE);
+
+  const mergedFilterQuery = getAnomaliesFilterQuery(
+    filterQuery,
+    anomaliesFilterQuery,
+    siemJobs,
+    anomalyScore,
+    flowTarget,
+    ip
+  );
+
+  return (
+    <>
+      <AnomaliesOverTimeQuery
+        endDate={endDate}
+        filterQuery={mergedFilterQuery}
+        sourceId="default"
+        startDate={startDate}
+        type={type}
+      >
+        {({ anomaliesOverTime, loading, id, inspect, refetch, totalCount }) => {
+          if (hideHistogramIfEmpty && !anomaliesOverTime.length) {
+            return <div />;
+          }
+
+          return (
+            <>
+              <AnomaliesOverTimeManage
+                data={anomaliesOverTime!}
+                endDate={endDate}
+                id={id}
+                inspect={inspect}
+                loading={siemJobsLoading || loading}
+                refetch={refetch}
+                setQuery={setQuery}
+                startDate={startDate}
+                totalCount={totalCount}
+                updateDateRange={updateDateRange}
+              />
+              <EuiSpacer />
+            </>
+          );
+        }}
+      </AnomaliesOverTimeQuery>
+      <AnomaliesTableComponent
+        startDate={startDate}
+        endDate={endDate}
+        skip={skip}
+        type={type as never}
```

Comment:
I think the type conversion here is not very intuitive,  so I tried to remove `as never` here.
I added a few lines to `public/containers/anomalies/anomalies_query_tab_body/types.ts`

```
type HostOrNetworkType =
  | HostsType.page
  | HostsType.details
  | NetworkType.page
  | NetworkType.details;


export type AnomaliesQueryTabBodyProps = QueryTabBodyProps & {
  startDate: number;
  endDate: number;
  skip: boolean;
  setQuery: SetQuery;
  narrowDateRange: NarrowDateRange;
  updateDateRange?: UpdateDateRange;
  anomaliesFilterQuery?: object;
  hideHistogramIfEmpty?: boolean;
  ip?: string;
  flowTarget?: FlowTarget;
  AnomaliesTableComponent: (
    props: HostOrNetworkProps & {
      hostName?: string;
      type: HostOrNetworkType;
      ip?: string;
      flowTarget?: FlowTarget;
    }
  ) => JSX.Element | null;
};
```

It is definitely not the best solution, but I kind of like it reflecting the current case. It's not necessary to change this unless you agree, but I'd love to learn what you think :)

Reply:
Thank you very much for the feedback, I have implemented as you suggested 💪